subprojects/ide-native/src/main/java/org/gradle/ide/xcode/internal/XcodeTarget.java

```diff
@@ -166,11 +167,15 @@ private static String toFileType(PBXTarget.ProductType productType) {
     public void addBinary(String configuration, Provider<? extends FileSystemLocation> outputFile, String architectureName) {
         binaries.add(new XcodeBinary(configuration, outputFile, architectureName));
         if (configuration.contains("Debug")) {
-            this.debugOutputFile = outputFile;
+            this.debugOutputFile.set(outputFile);
         }
     }
 
     public Property<String> getDefaultConfigurationName() {
         return defaultConfigurationName;
     }
+
+    public boolean isBuildable() {
+        return productType != null;
```

Comment:
This seems a little soft and potentially a defect waiting to happen.  For instance, what if someone adds an XCodeTarget with type `INDEXER`?  It might be better to push the buildability down into the type (i.e. `productType.isBuildable()`)

Reply:
Kind of. This type and how it's being used is internal. So we shouldn't expect someone from adding custom target. However, you are right we can push this down into `ProductType`. At the same time, we shouldn't allow someone to add an "indexer" as a target. It's kind of implicit at the moment until we find a way to merge both concepts together so we don't need an indexer target together with a buildable target.

Reply:
Let me think about this a bit more.

Reply:
Just to be clear, I was not referring to a user creating a target, I was talking about some sort of refactoring where we made a change and there's now a target with an indexer type.  Point is, whether a productType is present or not does not indicate that it is buildable - it's that only _certain_ types are buildable - in fact, it looks like there are a bunch of types in there that I don't think we even use.  I think we should make this a little tighter and a little more resilient to future changes, but I wouldn't waste a lot of time on it.

Reply:
I changed the logic to say it's unbuildable if there's no binaries.